audio-worklet/src/noisegate-demo.js

```diff
@@ -0,0 +1,284 @@
+          max: this.thresholdMax_,
+          step: 1,
+          default: this.threshold_,
+          name: 'Threshold (dBFS)'
+        });
+    this.thresholdSlider_.disable();
+
+    this.attackSlider_ =
+        new ParamController(containerId, this.setAttack.bind(this), {
+          type: 'range',
+          min: this.attackMin_,
+          max: this.attackMax_,
+          step: 0.01,
+          default: this.attack_,
+          name: 'Attack (s)'
+        });
+    this.attackSlider_.disable();
+
+    this.releaseSlider_ =
+        new ParamController(containerId, this.setRelease.bind(this), {
+          type: 'range',
+          min: this.releaseMin_,
+          max: this.releaseMax_,
+          step: 0.01,
+          default: this.release_,
+          name: 'Release (s)'
+        });
+    this.releaseSlider_.disable();
+
+    this.gainSlider_ =
+        new ParamController(containerId, this.setGain.bind(this), {
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.01,
+          default: this.masterGain_.gain.value,
+          name: 'Master Volume'
+        });
+
+     this.noiseGainSlider_ =
+        new ParamController(containerId, this.setNoiseGain.bind(this), {
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.01,
+          default: 1,
+          name: 'Noise Volume'
+        });
+
+    // Sound is processed by noise gate only if that radio button is selected.
+    document.getElementById(noisegateId)
+        .addEventListener('click', this.noisegateSelected.bind(this));
+    document.getElementById(withoutNoisegateId)
+        .addEventListener('click', this.withoutNoisegateSelected.bind(this));
+
+    // TODO: Allow user to switch between audio worklet and noise gate.
+  }
+
+  noisegateSelected() {
+    const t = this.context_.currentTime + 0.01;
+    this.noisegateGain_.gain.setValueAtTime(1, t);
+    this.withoutNoisegateGain_.gain.setValueAtTime(0, t);
+
+    if (this.playing) {
+      this.attackSlider_.enable();
+      this.releaseSlider_.enable();
+      this.thresholdSlider_.enable();
+    }
+  }
+
+  withoutNoisegateSelected() {
+    const t = this.context_.currentTime + 0.01;
+    this.withoutNoisegateGain_.gain.setValueAtTime(1, t);
+    this.noisegateGain_.gain.setValueAtTime(0, t);
+
+    this.attackSlider_.disable();
+    this.releaseSlider_.disable();
+    this.thresholdSlider_.disable();
+  }
+
+  /**
+   * Change the threshold.
+   * This is bound to an event listener by a ParamController.
+   * @param {Number} value The new threshold.
+   */
+  setThreshold(value) {
+    // TODO: When the param-controller update lands from the synthesizer branch,
+    // the stringToNumber conversion in this and subsequent methods should be
+    // removed.
+    const numericValue = parseInt(value);
+    if (numericValue < -100) throw 'The minimum threshold rate is -100.';
+    if (numericValue > 0) throw 'The maximum threshold rate is 0.';
```

Comment:
Are these things possible?  Doesn't the slider limit the values?

Probably didn't notice this in the noisegate design, but if there are limits on the threshold, shouldn't the noisegate implementation enforce those itself?